### PR TITLE
CORTX-32461: Add support for 1N and Multi-Node in Community Jenkins pipeline

### PR DESCRIPTION
**Pull request by shailesh-vaidya**
# Problem Statement
- Add support for 1N and Multi-Node in Community Jenkins pipeline 

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM
- [x] Jenkins pipelines used for testing
        1 Node - https://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/sv_space/job/community-deploy/61/parameters/
         3 Node ( 1+3 Nodes ) - https://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/sv_space/job/community-deploy/62/parameters/   
          9 Node ( 1+9 Nodes ) - https://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/sv_space/job/community-deploy/63/console

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] Jira ID/ COMMUNITY is prefixed in PR title
      Prefix Syntax - 
        `CORTX-<5 digit JIRA ID>: <PR Title>` - for Seagate Employees
        `COMMUNITY: <PR Title>`  - for open source contributors
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: jenkins/deployment/community/cortx-community-deploy.groovy [hunk 1]
```diff
@@ -200,31 +223,33 @@ pipeline {
...
                     ICON = 'warning.gif'
                     STATUS = 'UNSTABLE'
                 }
 
+                clusterStatusHTML  = "<pre>${CommunityBuildStatus}</pre>"
+
                 // Email Notification
                 env.build_stage = "${build_stage}"
-                env.cluster_status = "${clusterStatusHTML}"
+                env.cluster_status = "${clusterStatusHTML }"
```
Comment (pankajgborole) [inline jenkins/deployment/community/cortx-community-deploy.groovy:247]: Pls remove the extra space before closing curly bracket

Comment (shailesh-vaidya) [reply jenkins/deployment/community/cortx-community-deploy.groovy]: Fixed please review again.

**Review by pankajgborole — DISMISSED** [review]
Approved Changes

File: jenkins/deployment/community/cortx-community-deploy.groovy [hunk 2]
```diff
@@ -55,139 +59,158 @@ pipeline {
...
         }
         stage('Configure network and storage') {
+
             steps {
                 script { build_stage = env.STAGE_NAME }
+                script {
+                    env.PUBLIC_IP_LIST = sh( script: '''
+                    cd solutions/community-deploy/cloud/AWS && terraform show -json terraform.tfstate | jq .values.outputs.aws_instance_public_ip_addr.value | jq .[] | tr -d '"'
+                    ''', returnStdout: true).trim()
+                    env.PRIMARY_PUBLIC_IP = sh( script: '''
```
[the remark this answers is not in the document]
Comment (shailesh-vaidya) [reply jenkins/deployment/community/cortx-community-deploy.groovy:97]: We can do that. Need to take it as an improvement.

File: jenkins/deployment/community/cortx-community-deploy.groovy [hunk 3]
```diff
@@ -55,139 +59,158 @@ pipeline {
...
                 sh label: 'Generate locally build cortx images on worker nodes', script: '''
                 pushd solutions/community-deploy/cloud/AWS
-                    export WORKER_IP=$(cat ip_public.txt | jq '.[0]','.[1]','.[2]','.[3]' | tr -d '",[]')
-                    export BUILD_NODE=$(cat ec2_hostname.txt | jq '.[0]'| tr -d '",[]')
-                    export CORTX_SERVER_IMAGE="${BUILD_NODE}:8080/seagate/cortx-rgw:2.0.0-0"
-                    export CORTX_DATA_IMAGE="${BUILD_NODE}:8080/seagate/cortx-data:2.0.0-0"
-                    export CORTX_CONTROL_IMAGE="${BUILD_NODE}:8080/seagate/cortx-control:2.0.0-0"
-                    for wp in $WORKER_IP;do ssh -i cortx.pem -o 'StrictHostKeyChecking=no' centos@${wp} "sudo docker pull '${CORTX_SERVER_IMAGE}' && sudo docker pull '${CORTX_DATA_IMAGE}' && sudo docker pull '${CORTX_CONTROL_IMAGE}'";done
-                    popd
+                    for ip in $PUBLIC_IP_LIST;do ssh -i cortx.pem -o 'StrictHostKeyChecking=no' centos@"${ip}" "sudo -- sh -c 'sed -i 's,cortx-docker.colo.seagate.com,${PRIMARY_PRIVATE_HOSTNAME}:8080,g' /etc/docker/daemon.json && systemctl restart docker && sleep 120'";done
```
[the remark this answers is not in the document]
Comment (shailesh-vaidya) [reply jenkins/deployment/community/cortx-community-deploy.groovy:165]: Nop. Job only supports CentOS 7.9 as of now. We need to fetch username dynamically for multi-os support

File: jenkins/deployment/community/cortx-community-deploy.groovy [hunk 4]
```diff
@@ -55,139 +59,158 @@ pipeline {
...
-                    ssh -i cortx.pem -o 'StrictHostKeyChecking=no' centos@"${PRIMARY_PUBLIC_IP}" 'pushd /home/centos/cortx-re/solutions/kubernetes && export CORTX_SERVER_IMAGE='${CORTX_SERVER_IMAGE}' && export CORTX_DATA_IMAGE='${CORTX_DATA_IMAGE}' && export CORTX_CONTROL_IMAGE='${CORTX_CONTROL_IMAGE}' && sudo env SOLUTION_CONFIG_TYPE='${SOLUTION_CONFIG_TYPE}' env CORTX_SERVER_IMAGE='${CORTX_SERVER_IMAGE}' env CORTX_CONTROL_IMAGE='${CORTX_CONTROL_IMAGE}' env CORTX_DATA_IMAGE='${CORTX_DATA_IMAGE}' env COMMUNITY_USE='${COMMUNITY_USE}' ./cortx-deploy.sh --cortx-cluster'
-                    popd
-            '''
+                    export CORTX_SERVER_IMAGE="${PRIMARY_PRIVATE_HOSTNAME}:8080/seagate/cortx-rgw:2.0.0-0"
+                    export CORTX_DATA_IMAGE="${PRIMARY_PRIVATE_HOSTNAME}:8080/seagate/cortx-data:2.0.0-0"
+                    export CORTX_CONTROL_IMAGE="${PRIMARY_PRIVATE_HOSTNAME}:8080/seagate/cortx-control:2.0.0-0"
+                    ssh -i cortx.pem -o 'StrictHostKeyChecking=no' centos@"${PRIMARY_PUBLIC_IP}" 'pushd /home/centos/cortx-re/solutions/kubernetes &&
+                    export CORTX_SERVER_IMAGE='${CORTX_SERVER_IMAGE}' &&
```
[the remark this answers is not in the document]
Comment (shailesh-vaidya) [reply jenkins/deployment/community/cortx-community-deploy.groovy:182]: Script did not work in testing if we did not export values. Need to test this.

**Review by gauravchaudhari02 — APPROVED** [review]

**Pull request merged**